internal/dataplane/translator/subtranslator/httproute.go

```diff
@@ -235,13 +267,21 @@ func translateHTTPRouteRulesMetaToKongstateService(
 		applyTimeoutToServiceFromHTTPRouteRule(&service, ruleMeta.Rule)
 	}
 
-	routes, err := translateHTTPRouteRulesMetaToKongstateRoutes(rulesMeta)
-	if err != nil {
-		return kongstate.Service{}, err
+	if expressionRoutes {
+		routes, err := translateSplitHTTPRouteMatchesToKongstateRoutesWithExpression(matchesWithPriorities)
+		if err != nil {
+			return kongstate.Service{}, err
+		}
+		service.Routes = make([]kongstate.Route, 0, len(routes))
+		service.Routes = append(service.Routes, routes...)
+	} else {
+		routes, err := translateHTTPRouteRulesMetaToKongstateRoutes(rulesMeta)
+		if err != nil {
+			return kongstate.Service{}, err
+		}
+		service.Routes = make([]kongstate.Route, 0, len(routes))
+		service.Routes = append(service.Routes, routes...)
```

Comment:
The routes seem to be created in `kongExpressionRouteFromHTTPRouteMatchWithPriority` so they are not shared with anything. I believe in that case we can just assign them to `services.Routes` without copying no? Or is there a reason for this copy?

Reply:
removed this copy.